### PR TITLE
Update startAddr/startSymbol option management

### PR DESCRIPTION
**Pull request by vcave**

### CODE REVIEW

**Review by jleidel — APPROVED** [review]
approving this: @rkabrick can you also review?

File: src/RevCPU.cc [hunk 1]
```diff
@@ -267,25 +267,47 @@ RevCPU::RevCPU( SST::ComponentId_t id, const SST::Params& params )
...
+      StartAddr = ResolvedStartSymbolAddr;
     }
+  } else { // A start address was provided ...
+    // ... check if a symbol was provided and is compatible with the start address ...
+    if ((IsStartSymbolProvided) && (ResolvedStartSymbolAddr != StartAddr)) {
+      // ... they are different, don't know the user intent so error out now
+      output.fatal(CALL_INFO, -1,
+                  "Error: start address and start symbol differ for core=%lu startAddr=%lu StartSymbol=%s ResolvedStartSymbolAddr=%lu\n",
+                    id, StartAddr, StartSymbol.c_str(), ResolvedStartSymbolAddr);
+    } // ... else no symbol provided, continue on with StartAddr as the target
```
Comment (rkabrick) [inline src/RevCPU.cc:306]: Can you modify your outputs to use the formatting macros (`PRIu64`, `PRIu32`, `PRIx64` etc.)?

```cpp
output.fatal(CALL_INFO, -1,
                  "Error: start address and start symbol differ for core=%" PRIu32 " startAddr=%" PRIx64 " StartSymbol=%s ResolvedStartSymbolAddr=%" PRIu64 "\n",
                    id, StartAddr, StartSymbol.c_str(), ResolvedStartSymbolAddr);
```

File: src/RevCPU.cc [hunk 2]
```diff
@@ -267,25 +267,47 @@ RevCPU::RevCPU( SST::ComponentId_t id, const SST::Params& params )
...
   }
 
+  output.verbose(CALL_INFO, 11, 0, "Start address is 0x%lx\n", StartAddr);
```
Comment (rkabrick) [inline src/RevCPU.cc:309]: PRIx64

File: src/RevCPU.cc [hunk 3]
```diff
@@ -2609,10 +2630,8 @@ void RevCPU::CheckBlockedThreads(){
 // ----------------------------------
 void RevCPU::SetupArgs(uint32_t ThreadIDToSetup, RevFeature* feature){
   auto Argv = Opts->GetArgv();
-  // setup argc
   Threads.at(ThreadIDToSetup)->GetRegFile()->SetX(RevReg::a0, Argv.size());
-  Threads.at(ThreadIDToSetup)->GetRegFile()->SetX(RevReg::a1, Mem->GetStackTop() + 60);
-  return;
+  Threads.at(ThreadIDToSetup)->GetRegFile()->SetX(RevReg::a1, Mem->GetStackTop()+60);
```
Comment (rkabrick) [inline src/RevCPU.cc:2634]: You may want to also set the `gp` register to the value of `loader->GetSymbolAddr("__global_pointer$")`

**Review by rkabrick — CHANGES_REQUESTED** [review]
- Update the format strings to use PRI macros
- Set the `gp` register in the SetupArgs

File: src/RevCPU.cc (hunk 1 above)
Comment (vcave) [reply src/RevCPU.cc]: fixed

File: src/RevCPU.cc (hunk 3 above)
Comment (vcave) [reply src/RevCPU.cc]: per our parallel discussion, this is already done in thread init

**Comment by jleidel** [comment]
good to merge

**Review by jleidel — APPROVED** [review]

**Pull request merged**